motionplan/kinematic.go

```diff
@@ -2,63 +2,11 @@ package motionplan
 
 import (
 	"math"
-	"strings"
 
-	"github.com/pkg/errors"
-	pb "go.viam.com/api/component/arm/v1"
 	"gonum.org/v1/gonum/floats"
 	"gonum.org/v1/gonum/num/quat"
-
-	"go.viam.com/rdk/referenceframe"
-	"go.viam.com/rdk/spatialmath"
 )
 
-// ComputePosition takes a model and a protobuf JointPositions in degrees and returns the cartesian position of the
-// end effector as a protobuf ArmPosition. This is performed statelessly without changing any data.
-func ComputePosition(model referenceframe.Frame, joints *pb.JointPositions) (spatialmath.Pose, error) {
-	if len(joints.Values) != len(model.DoF()) {
-		return nil, errors.Errorf(
-			"incorrect number of joints passed to ComputePosition. Want: %d, got: %d",
-			len(model.DoF()),
-			len(joints.Values),
-		)
-	}
-
-	pose, err := model.Transform(model.InputFromProtobuf(joints))
-	if err != nil {
-		return nil, err
-	}
-
-	return pose, nil
-}
-
-// ComputeOOBPosition takes a model and a protobuf JointPositions in degrees and returns the cartesian
-// position of the end effector as a protobuf ArmPosition even when the arm is in an out of bounds state.
-// This is performed statelessly without changing any data.
-func ComputeOOBPosition(model referenceframe.Frame, joints *pb.JointPositions) (spatialmath.Pose, error) {
```

Comment:
This didn't actually get re-added to `referenceframe`

Reply:
I preserved the functionality in `ComputePosition` and deleted the old version of `ComputePosition` as it didnt do anything different from `model.Transform`

Reply:
There was a reason for this.
Nominally, `(something, error)` functions are supposed to return a good value and nil, or nothing and an error. `Transform` breaks this paradigm because an OOB call will return a (maybe) usable value AND an error. Checks of `if err != nil` therefore had to be aware of this.

Having both of these functions allows the user to select whether they want OOB to be considered an error, and have the function return canonical values.

Reply:
The logic is pretty simple I think.  If there is a non-nil error, then error.  If there is not then don't.  ComputePosition then becomes a way to call Transform and not get an error which means the value is usable

Reply:
If we do not like the fact that transform returns `somevalue, err` then we should modify the function itself since it is publicly accessible and therefore has the potential to be misused

Reply:
> ComputePosition then becomes a way to call Transform and not get an error which means the value is usable

Yes, exactly, that was its utility, which is being removed. Forcing a user to always do parsing on error checks will cause bugs in client code.

This change is actually worse than that; one function (`ComputePosition`) is being removed, and another function (`ComputeOOBPosition`) is being renamed to have the same name as the removed function, but with different functionality. So if some client Go code uses this function, they cannot simply update to the new package name, they need to actually go write their own function now if they don't want behavior to silently change.

Reply:
They don't need to do parsing on error checks.  If you call Transform and you get an error value it should be treated as an error.  `ComputePosition` is what you would call to not have to worry about that error checking.  `ComputeOOBPosition` is a poorly named function IMO since it implies it should be used on only OOB values which is not how it is used in practice.  

I don't want to be beholden forever to decisions we made over a year ago because someone might hypothetically be using a public function in a client script somewhere so that argument is not compelling to me.  Thats actually my motivation for wanting to make this change.  Having 3 ways to compute the position as public methods is not a good API design.  This change would cut it to two